File: swift/numbers/Tests/numbers_recursive_test/numbers_recursive_test.swift
import XCTest
@testable import numbers_recursive

final class RecursiveTests: XCTestCase {
    func testFibonacci() {
        XCTAssertEqual(NumbersRecursive.fibonacci(5), 5)
    }
    func testFactorial() {
        XCTAssertEqual(NumbersRecursive.factorial(5), 120)
    }
    func testSumNumbers() {
        XCTAssertEqual(NumbersRecursive.sumNumbers(5), 15)
    }
}

File: _experimental/_go/numbers/tests/numbers_recursive_test.go
package tests

import (
	"testing"

	. "example.com/numbers/numbers_recursive"
)

func TestFibonacci(t *testing.T) {
	result := Fibonacci(5)
	expected := 5
	if result != expected {
		t.Errorf("Fibonacci(5) = %d; want %d", result, expected)
	} else {
		t.Log("Fibonacci... [OK]")
	}
}

func TestFactorial(t *testing.T) {
	result := Factorial(5)
	expected := 120
	if result != expected {
		t.Errorf("Factorial(5) = %d; want %d", result, expected)
	} else {
		t.Log("Factorial... [OK]")
	}
}

func TestSumNumbers(t *testing.T) {
	result := SumNumbers(5)
	expected := 15
	if result != expected {
		t.Errorf("SumNumbers(5) = %d; want %d", result, expected)
	} else {
		t.Log("SumNumbers... [OK]")
	}
}
